.circleci/config.yml

```diff
@@ -528,7 +470,9 @@ jobs:
     parallelism: 4
     steps:
       - prepare_build_dir
+      - checkout
       - prepare_go
+      - generic_build
```

Comment:
It appears that we _always_:
```
- prepare_build_dir
- checkout
- prepare_go
- generic_build
```
now.  Can we simplify the yml?

Reply:
@jannotti https://github.com/algorand/go-algorand/pull/4426/commits/eca634d92c2586b9188d4f63eeb0cf80c8b08a79 consolidates `generic_build` pre-requisite steps.  Can you resolve if it looks good to you?

It's a reasonable callout.  I deliberately presented the PR with minimal topology changes thinking it'd simplify review.  Though I'm happy to fulfill the request.